planet/api/http.py

```diff
@@ -13,135 +13,176 @@
 # the License.
 
 """Functionality to perform HTTP requests"""
-
-# import http.client as http_client
+import asyncio
 import logging
-import os
-import re
-import time
 
-from requests import Session
-from requests.compat import urlparse
+import httpx
 
 from . import exceptions, models
 from . __version__ import __version__
 
-
-LOGGER = logging.getLogger(__name__)
-
-DEFAULT_CHUNK_SIZE = 32 * 1024
-
-USE_STRICT_SSL = not (os.getenv('DISABLE_STRICT_SSL', '').lower() == 'true')
-
+RETRY_COUNT = 5
 RETRY_WAIT_TIME = 1  # seconds
 
+LOGGER = logging.getLogger(__name__)
 
-# def setup_logging():
-#     log_level = LOGGER.getEffectiveLevel()
-#     urllib3_logger = logging.getLogger(
-#         'requests.packages.urllib3')
-#     urllib3_logger.setLevel(log_level)
-#
-#     # if debug level set then its nice to see the headers of the request
-#     if log_level == logging.DEBUG:
-#         http_client.HTTPConnection.set_debuglevel(1)
-#     else:
-#         http_client.HTTPConnection.set_debuglevel(0)
-#
 
-def _log_request(req):
-    LOGGER.info('%s %s %s %s', req.method, req.url, req.params, req.data)
+class Session():
+    '''Context manager for asynchronous communication with the Planet server.
```

Comment:
What are thoughts around something like `s/Planet servers?/Planet APIs/gi`?

Reply:
Hmm... I consider API to be a definition of an interface, whereas a server is the thing hosting the API. So we don't communicate with an API, we communicate with a server.

Reply:
Okay. I was curious because the language in the README and examples mentions APIs and not servers and so I wanted to ask because I found it a little jarring to read documentation and go "wait, is there a `planet.Server` I missed?!". :)

Reply:
Good to know! I will keep this in mind when I revisit the documentation in #242. Confusion is always good to note, thanks for the heads up.

Reply:
the colloquial usage of "Planet server" here is fine and matches the voice of our documentation elsewhere.